apps/frontend/src/components/contract-visualizer/index.tsx

```
@@ -673,7 +673,10 @@ export function ContractVisualizer() {
     setEdges(layoutedEdges);
     setLayouting(false);
     setInitialized(true);
-  }, [
+  },
+    // setNodes/setEdges are stable React Flow setters; this layout reset only tracks chain changes.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

ESLint

[error] Definition for rule 'react-hooks/exhaustive-deps' was not found.

(react-hooks/exhaustive-deps)